### PR TITLE
Restore tab names

### LINKED ISSUE #572: Incorrect tab names on restore
Restoring two tabs with different locations results in both tabs having the name of the last restored.

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/61513287-incorrect-tab-names-on-restore?utm_campaign=plugin&utm_content=tracker%2F65602118&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F65602118&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by jeremypw**
Fixes #572 

Refactors the setting of tab labels.  Each ViewContainer is given a unique id on construction which is used to identify the calling tab in "check_for_same_name" function - using current_tab is unreliable during restoring.

### CODE REVIEW

**Comment by danrabbit** [comment]
I can confirm that this fixes issues I'm having in master with restoring tab names, but I'm getting a lot of these criticals:

```
** (io.elementary.files:4010): CRITICAL **: 09:03:58.527: marlin_view_view_container_get_tab_name: assertion 'self != NULL' failed

** (io.elementary.files:4010): CRITICAL **: 09:03:58.527: marlin_view_window_set_title: assertion 'title != NULL' failed
```

**Comment by jeremypw** [comment]
Thanks - maybe the merges from master caused the error messages - I don't remember them when pushing - I'll look into it.

File: src/View/ViewContainer.vala [hunk 1]
```diff
@@ -368,15 +381,13 @@ namespace Marlin.View {
                     } catch (ConvertError e) {}
 
                     if (tab_name == null) {
-                        tab_name = Path.get_basename (slot_path);
+                        tab_name = location.get_parse_name ();;
```
Comment (danrabbit) [inline src/View/ViewContainer.vala]: two semicolons

**Comment by jeremypw** [comment]
Sorry about that!

**Pull request merged**